### PR TITLE
feat: puya debugging support

### PR DESCRIPTION
**Pull request by aorumbayev**
## Proposed Changes

- Deprecating `persist_sourcemaps`

### CODE REVIEW

File: src/algokit_utils/_debugging.py [hunk 1]
```diff
@@ -174,8 +175,18 @@ def _build_avm_sourcemap(  # noqa: PLR0913
     return AVMDebuggerSourceMapEntry(str(source_map_output_path), program_hash)
 
 
+@deprecated(
```
Comment (neilcampbell) [inline src/algokit_utils/_debugging.py:178]: In the case you want to debug TEAL, this is still useful. I thought we were only planning to remove the creation of the registry file?

File: src/algokit_utils/_debugging.py [hunk 2]
```diff
@@ -174,8 +175,18 @@ def _build_avm_sourcemap(  # noqa: PLR0913
...
+    version="3.0.0",
+)
 def persist_sourcemaps(
-    *, sources: list[PersistSourceMapInput], project_root: Path, client: "AlgodClient", with_sources: bool = True
+    *,
+    sources: list[PersistSourceMapInput],
+    project_root: Path,
+    client: "AlgodClient",
+    with_sources: bool = True,
+    persist_mappings: bool = False,
```
Comment (neilcampbell) [inline src/algokit_utils/_debugging.py:189]: Is there any need to have a flag that turns on this behaviour?

File: src/algokit_utils/_debugging.py (hunk 1 above)
Comment (aorumbayev) [reply src/algokit_utils/_debugging.py]: @neilcampbell probably got sidetracked with the debug package on utils side and overlooked updating py stuff again. Should be addressed now

File: src/algokit_utils/_debugging.py (hunk 2 above)
Comment (aorumbayev) [reply src/algokit_utils/_debugging.py]: Nope, good point - removed

**Review by neilcampbell — APPROVED** [review]

**Pull request merged**